### PR TITLE
Include term Attribute in Violations and Warnings Output

### PR DESCRIPTION
**Pull request by Arpit529Srivastava**
issue #1668 
## About
This Pull Request modifies the output format of the ec validate image command to include the term attribute in the violations and warnings reported through YAML or JSON formats. This change ensures that users can view important contextual information alongside each violation and warning without needing to provide the --info flag.

### Before:
Violations and warnings were reported without the term attribute by default, making it harder for users to understand the context and significance of each issue.
Users had to remember to use the --info flag to access additional details, which could lead to confusion or oversight.

### After:
Enhances Output: Automatically includes the term attribute in the JSON/YAML output for all violations and warnings, providing users with more context about each issue.
Improves Usability: Eliminates the need for the --info flag to access the term, making the output more user-friendly and informative by default.
Facilitates Filtering: Users can utilize the term for include/exclude causes more effectively, aiding in better decision-making during validation processes.

### CODE REVIEW

**Comment by Arpit529Srivastava** [comment]
@simonbaird @ralphbean @caugello @zregvart

 Please review this and add the label ```hacktoberfest-accepted``` and merge the pr. If any issue reach out to me.

File: internal/output/output.go [hunk 1]
```diff
@@ -217,9 +217,12 @@ func keepSomeMetadata(results []evaluator.Result) {
 	}
 }
 
+// keepSomeMetadataSingle retains only specific metadata keys in the result.
+// **Updated to include "term" by default as per the acceptance criteria.**
```
Comment (lcarva) [inline internal/output/output.go:221]: Let's remove this line.

File: internal/output/output.go [hunk 2]
```diff
@@ -217,9 +217,12 @@ func keepSomeMetadata(results []evaluator.Result) {
...
 func keepSomeMetadataSingle(result evaluator.Result) {
 	for key := range result.Metadata {
-		if key == "code" || key == "effective_on" {
+		// Retain "code", "effective_on", and "term" keys
```
Comment (lcarva) [inline internal/output/output.go:224]: This comment doesn't seem very helpful. Let's drop it.

**Comment by lcarva** [comment]
Thanks! Looks reasonable. Just a couple of minor comments. Could you also update the git commit with a more specific title and description?

Let's also update the tests as needed. See the Testing section in the README for how to run those locally if you want.

**Comment by Arpit529Srivastava** [comment]
@lcarva please let me know if there is any issue or any changes that you want

**Comment by Arpit529Srivastava** [comment]
> Thanks! Looks reasonable. Just a couple of minor comments. Could you also update the git commit with a more specific title and description?
> 
> Let's also update the tests as needed. See the Testing section in the README for how to run those locally if you want.

@lcarva please add label ```hactoberfest``` and ```hactoberfest-accepted```

**Comment by Arpit529Srivastava** [comment]
@lcarva can you please merge it

**Comment by lcarva** [comment]
@Arpit529Srivastava, please see my comments and the failed test.

**Comment by simonbaird** [comment]
+1 to fixing the commit message so it describes the change. https://cbea.ms/git-commit/ is a useful guide to follow.

**Comment by zregvart** [comment]
Apart from a nicer commit message and making sure tests pass, we could also update the text output? A change to the `internal/applicationsnapshot/templates/_results.tmpl` would be needed.

**Comment by Arpit529Srivastava** [comment]
Sure will do it by today

**Comment by lcarva** [comment]
Closing this due to lack of activity. Please, feel free to re-open it if you get back to it!

**Pull request closed without merging**